File: src/main/java/com/raihanorium/springreact/service/VoucherService.java
package com.raihanorium.springreact.service;

import com.raihanorium.springreact.model.Voucher;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Optional;

public interface VoucherService {
    Optional<Voucher> findById(Long id);

    Page<Voucher> findAll(Pageable pageable);

    Voucher save(Voucher voucher);
}

File: src/main/frontend/src/service/VoucherService.ts
import {Page} from "../model/Page";
import {Voucher} from "../model/Voucher";

export interface VoucherService {
    getVouchers(page?: number): Promise<Page<Voucher>>;
    getVoucher(id?: number): Promise<Voucher>;
    saveVoucher(voucher: Voucher): Promise<Voucher>;
}